### DISCUSSION TITLE
No genes found, skipping record

### DISCUSSION THREAD

**Discussion by Bio-finder** [antismash/antismash#681] [Q&A] [answered]
Hello,

I have currently an issue with antismash, all my records are CDS and I would like to annotate them with antismash. But when I run it in debug mode I get the following output:

`DEBUG    05/12 15:00:45   Stripping antiSMASH features and annotations from record: 1_11
DEBUG    05/12 15:00:45   Stripping antiSMASH features and annotations from record: 1_14
DEBUG    05/12 15:00:45   Stripping antiSMASH features and annotations from record: 1_16
DEBUG    05/12 15:00:45   Stripping antiSMASH features and annotations from record: 1_18
DEBUG    05/12 15:00:45   Stripping antiSMASH features and annotations from record: 1_20
`
...

`INFO     05/12 15:00:46   No genes found, skipping record
INFO     05/12 15:00:46   No genes found, skipping record
INFO     05/12 15:00:46   No genes found, skipping record
INFO     05/12 15:00:46   No genes found, skipping record
INFO     05/12 15:00:46   No genes found, skipping record
INFO     05/12 15:00:46   No genes found, skipping record
ERROR    05/12 15:00:46   all records skipped`

So why are my CDS not recognised as genes? Should I include something in the header of the fasta?

**Comment by kblin** [comment]
What do you mean by "all my records are CDS"? Are you trying to run antiSMASH on single-gene sequences?

The command line option you want to look at for making antiSMASH run gene finding is `--genefinding-tool`. Use `prodigal` for bacterial sequences and `glimmerhmm` for fungi. Having said that, especially for fungi you probably want to run a dedicated gene finding tool and then pass those found genes into antiSMASH.

**Comment by Bio-finder** [comment]
I already detected my genes and that's those records which are being skipped. That's why I don't understand the output 'no genes found' because they should all be genes. I put `--genefinding-tool none`

**Comment by Bio-finder** [comment]
I tried by giving the contigs and a gff file and this time it works! I thought that giving single gene sequence would be the easiest but it seems that it's not compatible with your tool?

**Comment by SJShaw** [comment, marked as answer]
As a genome mining tool, antiSMASH requires genomic context for any meaningful results. As such, protein inputs are invalid and records that are a single gene without gene annotations are also invalid.
